### DISCUSSION TITLE
Initial Reflections

### DISCUSSION THREAD

**Discussion by Brennan1994** [HydrologicEngineeringCenter/AEP-Grid-Generator#1] [Show and tell]
The AEP-Grid-Generator so far provides examples and proves we can do a few things

- Extract data from an HDF file using the RAS APIs.
- Write data to an HDF file using the RAS APIs.
- View the overwritten data in RASMapper. 
- Produce reasonable looking gridded data representing quantile AEP events. 

This is pretty cool. 

Potential obstacles
- Requires a whole realization of RAS output be stored to compute.  

Weaknesses

- Currently we're tied to reading from and writing to RAS .hdf. We could abstract this a bit to also work with reading and writing to gridded data if we wanted. Or reading from RAS and writing to Grids. All possible.

**Comment by eheisman** [comment]
Nice to have: Storing with a name.  Likely requires RASMapper to rework data storage internal to `.hdf` file.  But this isn't a major concern now.

Next steps:
- configuration object
- add logic to pull values from AEP results from the 'per realization' dataset.
- use a model with multiple realizations to build a process to collect AEP results across realizations into confidence bounds.
- script to launch AEP Grid Generator during compute.
- tell RASMapper to output tiffs with AEP labels for use in GIS.
